### PR TITLE
Red Black Trees in java

### LINKED ISSUE #768: Red_Black_Tree in java
Would like to write java implementation of Red Black Trees

### PR DESCRIPTION
**Pull request by aneesh98**
Fixes #768
#### Checklist:
 [x] 4 space indentation.
 [x] Coding conventions are followed.
 [x] Sample Input / Output.
 [x] Logic Documentation (Comments).
 [x] File names are correct.

#### Changes proposed in this pull request:
- Adding Red Black Trees Implementation in java

#### Languages Used:
- Java

#### Files Added
- Red_Black_Tree/RedBlackTree.java

### CODE REVIEW

**Comment by marwahmanbir** [comment]
You have removed the checklist from the PR template, please add and complete the checklist.

**Comment by marwahmanbir** [comment]
Also the "Files Added" section

**Comment by aneesh98** [comment]
I've done the changes

File: Red_Black_Tree/RedBlackTree.java [hunk 1]
```diff
@@ -0,0 +1,331 @@
+import java.util.*;
+/*
+RED BLACK TREE IN JAVA
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Put this at the top of the file as a single comment.

File: Red_Black_Tree/RedBlackTree.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+*/
+
+
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Remove unnecessary blank lines.

File: Red_Black_Tree/RedBlackTree.java [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    {
+        Node node=new Node();
+        node.data=data;
+        node.color=color;
+        node.left=nullNode(node);
+        node.isNull=false;
+        node.right=nullNode(node);
+        if(parent!=null)
+        {
+            node.parent=parent;
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Add a space around all binary operators except " . "
Never add space around unary operators, increment (++) and decrement (--) operators.

File: Red_Black_Tree/RedBlackTree.java [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+    {
+        Node leaf=new Node();
+        leaf.color=BLACK;
+        leaf.isNull=true;
+        leaf.parent=parent;
+        return leaf;
+    }
+    public void rotate(Node root, boolean changeColor, String type)
+    {   
+        /*
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Add a blank line before every comment to improve readability.

File: Red_Black_Tree/RedBlackTree.java [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+                root.left = parent;
+                parent.parent = root;
+                parent.right = left;
+                if(left!=null)
+                {
+                    left.parent = parent;
+                }
+                if(changeColor)
+                {
+                    root.color= BLACK;
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Fix spacing around operators.

File: Red_Black_Tree/RedBlackTree.java [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+        10 B
+                            5 B
+        */
+        System.out.println("Inorder traversal:");
+        redBlackTree.printInOrder(root);
+        /* Inorder traversal:
+           5B 10B 15R 18B 25R 28R 50R 55B 76R */
+        System.out.println();
+    }
+}
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Add after line 331 in order:
- Single blank line
- Sample input-output
- Single blank line

File: Red_Black_Tree/RedBlackTree.java [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+        if(root == null || root.isNull) {
+            return;
+        }
+        displayRedBlackTree(root.right, space + 20);
+        for(int i=0; i < space; i++) {
+            System.out.print(" ");
+        }
+        System.out.println(root.data + " " + (root.color == BLACK ? "B" : "R"));
+        
+
```
Comment (marwahmanbir) [inline Red_Black_Tree/RedBlackTree.java]: Remove unnecessary blank lines.

**Comment by aneesh98** [comment]
Please check the latest commit if its satisfactory.

**Pull request merged**